runtime/ast/expression.go

```diff
@@ -220,6 +220,68 @@ func (*StringExpression) precedence() precedence {
 	return precedenceLiteral
 }
 
+// StringTemplateExpression
+
+type StringTemplateExpression struct {
+	Values      []string
+	Expressions []Expression
```

Comment:
How are string templates represented by these fields? Is the assumption that they are interleaved, i.e., `V[0] + E[0] + ... + E[N-1] + V[N]`? So for example, 
- would `"a\(b)c"` be `Values: []string{"a", "c"}, Expressions: []Expression{b}`, and
- would `"\(a)b\(c)"` be `Values: []string{"", "b", ""}, Expressions: []Expression{a, c}`? 

Could you please add some more tests for these and similar cases?

Reply:
> Is the assumption that they are interleaved, i.e., `V[0] + E[0] + ... + E[N-1] + V[N]`?

Yes they are assumed to be interleaved and this is enforced by the parser. 

> Could you please add some more tests for these and similar cases?

Added the above tests and one more.

Reply:
Makes sense! Can you please leave an explanation in the code, maybe also with these examples?

It might also be good to assert this structure in the constructor (`NewStringTemplateExpression`), to prevent illegal state